### ISSUE TITLE
Implement early stopping

### ISSUE THREAD

**Issue by RAMitchell** [rapidsai/legate-boost#18]
Via customisable callback, see XGBoost for an example.

The metrics interface can be modified, adding a method to indicate if training loss is expected to increase or decrease during training.
